### PR TITLE
Fix dead proxy not being rotated

### LINKED ISSUE #2149: Account not released when removing inactive proxy
## Expected Behavior
After proxy refresh, if a proxy is deleted from proxy list (inactive, banned...), account should start a new login with a new proxy.

## Current Behavior
Account doesn't get released, worker stays forever in "logging into account..." but log says "Account xxx proxy xxx is not in a live list any more".

## Possible Solution
Maybe set the account as failed?

## Steps to Reproduce (for bugs)
1. Load a proxy list
2. Block the IP of a proxy using a firewall (Windows firewall works)
3. Wait

## Context
Workerks get stuck as proxies fail, even if they come up some time after.

## Your Environment
* Version used: Last dev
* Environment name and version (e.g. Python 2.7): Python 2.7
* Operating System and version (desktop or mobile): Windows 10

### PR DESCRIPTION
**Pull request by AEtHeLsYn**
## Description
Fix to remove proxy from account if proxy is dead

## Motivation and Context
Fixes #2149

## How Has This Been Tested?
Local map

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by AEtHeLsYn** [comment]
@sebastienvercammen If no rotation is set we should avoid the search thread to get stuck with a dead proxy, we must force a new one.

File: pogom/account.py [hunk 1]
```diff
@@ -63,6 +61,10 @@ def setup_api(args, status, account):
         api.set_proxy({
             'http': status['proxy_url'],
             'https': status['proxy_url']})
+        if (status['proxy_url'] not in args.proxy):
+            log.warning(
+                'Proxy %s is not active but proxy rotation is set to none.',
+                status['proxy_url'])
```
Comment (sebastienvercammen) [inline pogom/account.py]: Rephrase:
```
Tried replacing proxy %s with a new proxy, but proxy rotation is disabled ("none"). If this isn't intentional, enable proxy rotation.
```

File: pogom/models.py [hunk 2]
```diff
@@ -2384,15 +2385,23 @@ def encounter_pokemon(args, pokemon, account, api, account_sets, status,
...
-            # these single-use variables are making me ill.
-            if args.hash_key:
-                key = key_scheduler.next()
-                log.debug('Using hashing key %s for this encounter.', key)
-                hlvl_api.activate_hash_server(key)
+        # If the already existent API is using a proxy but
+        # it's not alive anymore, we need to get a new proxy.
+        elif (args.proxy and
+              (hlvl_api._session.proxies['http'] not in args.proxy)):
+                proxy_num, proxy_new = get_new_proxy(args)
```
Comment (sebastienvercammen) [inline pogom/models.py]: Readability/consistency:

> proxy_num

to:

> proxy_idx

File: pogom/models.py [hunk 3]
```diff
@@ -2384,15 +2385,23 @@ def encounter_pokemon(args, pokemon, account, api, account_sets, status,
...
+                hlvl_api.set_proxy({
+                    'http': proxy_new,
+                    'https': proxy_new})
```
Comment (sebastienvercammen) [inline pogom/models.py]: You also need to set the new proxy on pgoapi's auth provider.

**Pull request closed without merging**